diff --git a/include/openvpn-plugin.h.in b/include/openvpn-plugin.h.in
index 05bffab..05b4b6a 100644
--- a/include/openvpn-plugin.h.in
+++ b/include/openvpn-plugin.h.in
@@ -169,7 +169,7 @@ typedef void *openvpn_plugin_handle_t;
 /*
  * We are compiling OpenVPN.
  */
-/* #define OPENVPN_PLUGIN_DEF        typedef */
+#define OPENVPN_PLUGIN_DEF        typedef
 #define OPENVPN_PLUGIN_FUNC(name) (*name)

 #else  /* ifdef OPENVPN_PLUGIN_H */
