--- i386/OemWin2k.inf	Mon Mar  5 23:17:34 2007
+++ amd64/OemWin2k.inf	Mon Mar  5 23:17:35 2007
@@ -61,9 +61,9 @@
 ;                      Manufacturer + Product Section (Done)
 ;----------------------------------------------------------------
 [Manufacturer]
-   %Provider% = tap0901
+   %Provider% = tap0901, NTamd64
 
-[tap0901]
+[tap0901.NTamd64]
    %DeviceDescription% = tap0901.ndi, tap0901
 
 ;---------------------------------------------------------------
